ls1intum/Artemis
# Show a link to the commit in the result details

### Checklist

### Motivation and Context
Currently, students cannot see what result belongs to which commit, especially in the code editor or exercise details. Usually, this is just the latest commit in the project, but it can be the case that either a corresponding result is not visible for them or has not been generated, see #3976.

### Description
Add a link to the top of the result details modal in case of programming exercises. We reuse the logic from the participation submission component here, and moved the commit link creation logic into the programming exercise utilities to avoid duplication.

In case the exercise is not available, we just show the commit in plain text. If the submission is not available, we do not display the note about the linked commit at all, as it is not possible then.

### Steps for Testing
1. Work with programming exercises: commit and wait for results. Some suggestions: try template, solution and test repository commits as well as regular student working in the code editor. It is also OK to use old results here.
2. Open the result details (click the result string)
3. You should see a message at the top of the feedback modal that links to a commit as shown in the screenshot.
4. Check that the link works.

*It might also be useful to test different locations where results can appear. It does currently not work for the assessment code editor.*

You can also do a quick check that it does not appear for other exercise results.

### Review Progress





- Code Review
- Manual Tests

### Test Coverage
* result-detail.component.ts 95.36%
  ![image](https://user-images.githubusercontent.com/11130248/132477583-7d708e31-2823-4e70-bbb6-7739dab5e41b.png)
* programming-exercise.utils.ts → full branch coverage of `createCommitUrl`
  ![image](https://user-images.githubusercontent.com/11130248/133668475-865e289e-f5b5-40ea-baa3-b0d64279cfe3.png)

### Screenshots
![image](https://user-images.githubusercontent.com/11130248/132369406-ac13874f-7739-45a0-9077-4faa166a8821.png)
